### PR TITLE
Port configuration incremental update support

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
HLD: https://github.com/Azure/SONiC/pull/985

**What I did**

1. portsyncd no longer handle port configuration change and put it to APP DB
2. Implement incremental configuration change in portmgr
3. Adjust portsorch to meet incremental configuration change requirement 

**Why I did it**

1. portsyncd and portmgrd both handles port configuration change, this causes unnecessary handling in ports orch
2. portmgrd shall do increment configuration update

**How I verified it**

1. manual test
2. regression test
3. added new unit test case

**Details if related**

### CODE REVIEW

**Comment by prsunny** [comment]
Could you please provide HLD or any issue submitted for this PR?

**Comment by liat-grozovik** [comment]
> Could you please provide HLD or any issue submitted for this PR?

@prsunny  the design was done over emails. We can share the detailed in the PR or sent it over google group. I don't think any change in the current design goes with an HLD. What do you say?

**Review by dgsudharsan — DISMISSED** [review]

**Comment by Junchao-Mellanox** [comment]
Wait for https://github.com/Azure/sonic-swss/pull/2195 to fix the DPB vs test failure.

**Comment by prgeor** [comment]
@Junchao-Mellanox are these changes in this [context](https://github.com/Azure/SONiC/pull/932#discussion_r807625174) ?

**Comment by Junchao-Mellanox** [comment]
> @Junchao-Mellanox are these changes in this [context](https://github.com/Azure/SONiC/pull/932#discussion_r807625174) ?

Yes

**Review by dgsudharsan — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@prgeor kindly reminder to review and signoff

**Comment by prgeor** [comment]
@Junchao-Mellanox could you please add reference to relevant HLD in the description?

**Comment by Junchao-Mellanox** [comment]
HLD: https://github.com/Azure/SONiC/pull/985

**Comment by Junchao-Mellanox** [comment]
@prgeor could you please review?

**Review by keboliu — DISMISSED** [review]

**Comment by prgeor** [comment]
@qiluo-msft could you review?

**Comment by prsunny** [comment]
IMO, this is a critical change and may have wider implications. IIRC, there are legacy reasons why portsyncd is listening to CONFIG_DB table. For e.g, some implementations configure kernel MTU differently than H/W MTU.  In such cases, if portsyncd is the source of truth, then H/W MTU will be same as kernel MTU.  Also, I don't see the HLD reviewed/signed-off. This needs to have some thorough review.

**Comment by prgeor** [comment]
> HLD: [Azure/SONiC#985](https://github.com/Azure/SONiC/pull/985)

I am not sure if auto-neg HLD would be the right place to put this design change which is not AN specific.

**Comment by liat-grozovik** [comment]
@prgeor kindly reminder to review and signoff

File: portsyncd/portsyncd.cpp [hunk 1]
```diff
@@ -135,28 +131,6 @@ int main(int argc, char **argv)
...
-                    string key = kfvKey(entry);
-
-                    if (port_cfg_map.find(key) != port_cfg_map.end())
-                    {
-                        /* For now we simply drop previous pending port config */
-                        port_cfg_map.erase(key);
-                    }
-                    port_cfg_map[key] = entry;
-                }
-                handlePortConfig(p, port_cfg_map);
```
Comment (Junchao-Mellanox) [inline portsyncd/portsyncd.cpp:159]: put a log here to prove that: one field change will trigger all port configuration change

**Comment by prsunny** [comment]
Discussion notes:
1. Split portsyncd change to another PR. 
2. In the same PR, change portmgr to add more attributes and remove config_cache from cgrmgr
3. For every port config change, portsyncd loops through the portcfg as the SELECT is invoked. Junchao to attach logs that shows any port attribute change shall invoke "handlePortConfig" function
4. Revisit the portsorch to only handle the final admin state with a m_config_admin_status.

**Comment by Junchao-Mellanox** [comment]
Close this. Will create new PR.

**Pull request closed without merging**